File: Source/Extensions/Tessa.Extensions.Default.Client/Workflow/WorkflowEngine/KrCheckStateTileManagerUIExtension.cs
using System;
using System.Linq;
using System.Threading.Tasks;
using Tessa.Extensions.Default.Shared.Workflow.WorkflowEngine;
using Tessa.Localization;
using Tessa.UI.Workflow;
using Tessa.Workflow.Helpful;

namespace Tessa.Extensions.Default.Client.Workflow.WorkflowEngine
{
    public sealed class KrCheckStateTileManagerUIExtension : IWorkflowEngineTileManagerUIExtension
    {
        private static readonly string extendedSectionName = WorkflowEngineHelper.GetExtendedSectionName("KrCheckStateTileExtension");

        /// <inheritdoc />
        public Guid ID => KrTileManagerExtensionsHelper.KrCheckStateExtensionID;

        /// <inheritdoc />
        public Task ModifyButtonRowAsync(IWorkflowEngineTileManagerUIExtensionContext context)
        {
            if (context.AllButtonRows.TryGetValue(extendedSectionName, out var contextRoleRows)
                && contextRoleRows.Count > 0)
            {
                if (context.Result.Length > 0)
                {
                    context.Result.AppendLine();
                }

                context.Result.Append(LocalizationManager.Localize("$KrTileExtensions_CheckState_States"));
                context.Result.Append(string.Join("; ", contextRoleRows.Select(x => LocalizationManager.Localize((string)x["StateName"]))));
            }

            return Task.CompletedTask;
        }
    }
}

File: WebClient SDK/src/default/ui/workflowEngine/krCheckStateTileManagerUIExtension.ts
import { LocalizationManager } from 'tessa/localization';
import {
  WorkflowEngineTileManagerUIExtension,
  IWorkflowEngineTileManagerUIExtensionContext
} from 'tessa/ui/workflow';

/**
 * Обработка расширения прав доступа к тайлам процесса, которое проверяет состояния документа.
 */
export class KrCheckStateTileManagerUIExtension extends WorkflowEngineTileManagerUIExtension {
  //region methods

  getExtensionId(): guid {
    return '9fce6311-1746-412a-9346-77394caebe91';
  }

  async modifyButtonRow(context: IWorkflowEngineTileManagerUIExtensionContext): Promise<void> {
    const contextRoleRows = context.allButtonRows.get('Extension_KrCheckStateTileExtension');

    if (contextRoleRows && contextRoleRows.length > 0) {
      const arrayValues: string[] = [];

      context.result += LocalizationManager.instance.localize(
        '$KrTileExtensions_CheckState_States'
      );
      contextRoleRows.map(x => {
        arrayValues.push(LocalizationManager.instance.localize(x.get('StateName')));
      });
      context.result += arrayValues.join('; ');
      context.result += '\n';
    }
  }

  //endregion
}
